extensions/amp-viewer-integration/0.1/test/integration/test-amp-viewer-integration.js

```diff
@@ -339,25 +339,21 @@ describes.sandboxed('AmpViewerIntegration', {}, (env) => {
 
         const resolveSpy = env.sandbox.stub();
         const rejectSpy = env.sandbox.stub();
-        const waitingForResponse = {
+        messaging.waitingForResponse_ = {
           '1': {
             resolve: resolveSpy,
             reject: rejectSpy,
           },
         };
 
-        env.sandbox
-          .stub(messaging, 'waitingForResponse_')
-          .callsFake(waitingForResponse);
         messaging.handleMessage_(event);
 
         expect(resolveSpy).to.have.been.calledWith(data);
       });
 
-      // TODO(chenshay, #12476): Make this test work with sinon 4.0.
-      it.skip('handleMessage_ should reject', () => {
+      it('handleMessage_ should reject', () => {
         const event = {
-          source: window,
+          source: env.win,
```

Comment:
I love this change. It prevents the pollution of the global `window` object.

Reply:
👍🏽